packages/@aws-cdk/aws-codepipeline-actions/lib/codebuild/build-action.ts

```diff
@@ -92,6 +92,21 @@ export class CodeBuildAction extends Action {
     this.props = props;
   }
 
+  /**
+   * Reference a CodePipeline variable defined by the CodeBuild project this action points to.
+   * Variables in CodeBuild actions are defined using the 'exported-variables' subsection of the 'env'
+   * section of the buildspec.
+   *
+   * @param variableName the name of the variable to reference.
+   *   A variable by this name must be present in the 'exported-variables' section of the buildspec
+   *
+   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html#build-spec-ref-syntax
+   */
+  public variable(variableName: string): string {
+    this.variableWasReferenced();
```

Comment:
I believe calling this can be avoided (and it's also a very error-prone API)

Reply:
Moved to what you suggested [here](https://github.com/aws/aws-cdk/pull/5604#discussion_r365583748).